### PR TITLE
Fix create_ak for ECC keys

### PR DESCRIPTION
**Pull request by ionut-arm**
Fixing the parameters for creating AKs in the Endorsement Hierarchy. The `count` value part of the `EccScheme` has been adjusted, and an empty `EccPoint` was added as unique identifier for the key.

### CODE REVIEW

**Review by wiktor-k — DISMISSED** [review]
Looks good :+1: 

Is there a test case for `EcDaa` already? (so that both codepaths are tested).

**Comment by ionut-arm** [comment]
I'm curious now why that `ecc_rsa` test was supposed to fail 🤔

**Comment by wiktor-k** [comment]
> EDIT: nevermind, noticed the scheme... why does it work with swtpm, though....

I think a better explanation in the test would be a good idea. Also, instead of manual panic: [`#[should_panic]`](https://doc.rust-lang.org/book/ch11-01-writing-tests.html#checking-for-panics-with-should_panic) maybe even with `expected`...

Just throwing some random work ideas your way :sweat_smile:

**Review by Superhepper — APPROVED** [review]

File: tss-esapi/tests/integration_tests/abstraction_tests/ak_tests.rs
Comment (Superhepper) [inline tss-esapi/tests/integration_tests/abstraction_tests/ak_tests.rs:45]: I don't understand this panic. Should you not test that the correct error is generated instead?

Comment (wiktor-k) [reply tss-esapi/tests/integration_tests/abstraction_tests/ak_tests.rs:45]: I guess Ionut did so because I suggested it in a previous comment. Sorry 🙇‍♂️! Maybe indeed `assert!(matches!(...` would be a better option 🤔

Comment (Superhepper) [reply tss-esapi/tests/integration_tests/abstraction_tests/ak_tests.rs:45]: I mean for me ```#[should_panic]``` is used to test that the library panics under certain conditions. And we try very hard to not panic any where in the library so that should hopefully not happen.

Things that one can do to test is:
1. If the error in it self is not that import but rather that an error is returned at all then you can:
```rust
let result = ak::foo();
assert!(result.is_err(), "Function fo should have returned an error.");
```
This is not done any where in our code currently because the returned errors are a part of the API.

2. The specific error is important:
```rust
let result = ak::foo();
if let Err(actual_error) = result {
    assert_eq!(
        Error::WrapperError(WrapperErrorKind::InvalidParam),
        actual_error,
        "Foo did not produce the expected error."
    );
} else {
    panic!("Foo is expected to return an error");
}
```
This is done extensively when testing the TSS return code errors.

Comment (ionut-arm) [reply tss-esapi/tests/integration_tests/abstraction_tests/ak_tests.rs:45]: Thanks for the pushback, should've thought this through 😅 

I'll tweak it and push back up.

**Comment by Firstyear** [comment]
What should we do to unblock this pr?

**Comment by Superhepper** [comment]
I do not think it is blocked I just think @ionut-arm has been busy and not been able to work on this.

**Comment by Superhepper** [comment]
@ionut-arm Tell me if you need any help with this. I can fix the tests for you if you like.

File: tss-esapi/tests/integration_tests/abstraction_tests/ak_tests.rs
Comment (ionut-arm) [reply tss-esapi/tests/integration_tests/abstraction_tests/ak_tests.rs]: Right, only took about 3 months, hope it works!

**Comment by ionut-arm** [comment]
> @ionut-arm Tell me if you need any help with this. I can fix the tests for you if you like.

ok, finally got to this... @Firstyear - apologies for the long wait, I reckon this is ready to go (pending approval, of course)

**Review by Superhepper — APPROVED** [review]

**Review by Firstyear — APPROVED** [review]

**Review by hug-dev — APPROVED** [review]

**Pull request merged**
